### PR TITLE
Basic Docs setup

### LINKED ISSUE #5: Add readthedocs.io documentation
We should have a publicly available documentation for the package and all its tools.

### PR DESCRIPTION
**Pull request by maxfischer2781**
This pull request adds basic documentation. Changes include:

* [x] basic sphinx docs,
* [x] automatic readthedocs builds.

There is a test build at [readthedocs](https://change-log.readthedocs.io/en/feature-docs/index.html). Closes #5.

### CODE REVIEW

File: docs/conf.py
Comment (eileen-kuehn) [inline docs/conf.py]: If you exchange those two lines you can even include the ``author`` into the copyright ;)

File: docs/subtool/log.rst
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
Each fragment file contains a mapping that must contain
```

File: docs/subtool/log.rst
Comment (eileen-kuehn) [inline docs/subtool/log.rst:29]: Please consider to put these options into the description text not just the example code.

File: docs/subtool/log.rst
Comment (eileen-kuehn) [inline docs/subtool/log.rst]: What do you think about using an example that does match this repository? For example you could include this PR itself. So if anybody tries to interpret this also finds the referenced PR and so on.

File: docs/subtool/log.rst
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
The ``version`` is optional and in most cases does not need manual definition.
```

File: docs/subtool/log.rst
Comment (eileen-kuehn) [inline docs/subtool/log.rst]:
```suggestion
Unversioned fragments belong to the *next* release, and
```

**Review by eileen-kuehn — CHANGES_REQUESTED** [review]
Great thing, we should consider having something like this in a project template :)

However, I left some minor comments you should have a look at before merging.

File: docs/subtool/log.rst
Comment (maxfischer2781) [reply docs/subtool/log.rst]: I think that's a great idea!

File: docs/subtool/log.rst
Comment (maxfischer2781) [reply docs/subtool/log.rst:29]: Documented the expected choices for ``category``.

File: docs/subtool/log.rst
Comment (maxfischer2781) [reply docs/subtool/log.rst]: Now using this PR and a new issue #5 as example.

File: docs/conf.py
Comment (maxfischer2781) [reply docs/conf.py]: Dome.

**Review by eileen-kuehn — APPROVED** [review]
Great, especially the self-reference 👍

**Pull request merged**
